mkosi/vmspawn.py

```diff
@@ -0,0 +1,227 @@
+        f"--qemu-mem={config.qemu_mem}",
+        f"--qemu-kvm={config_feature_to_tristate(config.qemu_kvm)}",
+
+        # we need to handle this ourselves as we cannot pass the fd to vmspawn
+        # directly but have to pass it to qemu instead
+        # to prevent a conflict we disable vsock for vmspawn
+        "--qemu-vsock=no",
+    ]
+
+    if config.qemu_gui:
+        cmdline.append("--qemu-gui")
+
+    if config.secure_boot:
+        cmdline.append("--secure-boot=yes")
+
+    for name, val in config.credentials.items():
+        cmdline.append(f"--set-credential={name}:{val}")
+
+    # values which we need to pass directly to qemu
+    qemu_cmdline: list[PathString] = []
+
+    if QemuDeviceNode.vhost_vsock in qemu_device_fds:
+        qemu_cmdline += [
+            "-device",
+            f"vhost-vsock-pci,guest-cid={machine_cid(config)},vhostfd={qemu_device_fds[QemuDeviceNode.vhost_vsock]}"
+        ]
+
+    with contextlib.ExitStack() as stack:
+        if config.ephemeral:
+            fname = stack.enter_context(copy_ephemeral(config, config.output_dir_or_cwd() / config.output))
+        else:
+            fname = config.output_dir_or_cwd() / config.output
+
+        # Make sure qemu can access the ephemeral copy. Not required for directory output because we don't pass that
+        # directly to qemu, but indirectly via virtiofsd.
+        os.chown(fname, INVOKING_USER.uid, INVOKING_USER.gid)
+
+        if config.output_format == OutputFormat.disk and config.runtime_size:
+            run(["systemd-repart",
+                 "--definitions", "",
+                 "--no-pager",
+                 f"--size={config.runtime_size}",
+                 "--pretty=no",
+                 "--offline=yes",
+                 fname])
+
+        root = None
+        if kernel:
+            qemu_cmdline += ["-kernel", kernel]
+
+            # We can't rely on gpt-auto-generator when direct kernel booting so synthesize a root=
+            # kernel argument instead.
+            root = finalize_root(find_partitions(fname))
+            if not root:
+                die("Cannot perform a direct kernel boot without a root or usr partition")
+
+        cmdline += [f"--image={fname}"]
+
+        if kernel and (KernelType.identify(kernel) != KernelType.uki or not config.architecture.supports_smbios()):
+            kcl = config.kernel_command_line + config.kernel_command_line_extra
+        else:
+            kcl = config.kernel_command_line_extra
+
+        if root:
+            kcl += [root]
+
+        for src, target in config.runtime_trees:
+            sock = stack.enter_context(start_virtiofsd(src, uidmap=True))
+            qemu_cmdline += [
+                "-chardev", f"socket,id={sock.name},path={sock}",
+                "-device", f"vhost-user-fs-pci,queue-size=1024,chardev={sock.name},tag={sock.name}",
+            ]
+            kcl += [f"systemd.mount-extra={sock.name}:{target or f'/root/src/{src.name}'}:virtiofs"]
```

Comment:
This should be supported natively in vmspawn. Probably named `--bind` (same as nspawn)

Reply:
Okay I will work on adding this to vmspawn.

Reply:
Fixed in 594d4fb.